Box.V2/Request/HttpRequestHandler.cs

```diff
@@ -82,12 +84,12 @@ public class HttpRequestHandler : IRequestHandler
                         ((response.StatusCode == TooManyRequests && !isMultiPartRequest)
                         ||
                         (response.StatusCode == HttpStatusCode.Accepted && retryAfterHeader != null)) 
-                        && numRetries-- > 0)
+                        && retryCounter++ < 3)
                     {
                         TimeSpan delay = TimeSpan.FromSeconds(2);
                         if (retryAfterHeader.Delta.HasValue)
                         {
-                            delay = retryAfterHeader.Delta.Value;
+                           delay = expBackoff.GetRetryTimeout(retryCounter, retryAfterHeader.Delta.Value);
```

Comment:
I worry a bit about using both the Retry-After header and the exponential backoff — what if in the future the Retry-After header starts giving higher values?  Then the customer's backoff time will go up really fast.